jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/HttpTransportOverHTTP2.java

```diff
@@ -352,117 +353,259 @@ public void abort(Throwable failure)
 
     private class TransportCallback implements Callback
     {
-        private State state = State.IDLE;
-        private Callback callback;
-        private Throwable failure;
-        private boolean commit;
+        private State _state = State.IDLE;
+        private Callback _callback;
+        private boolean _commit;
+        private Throwable _failure;
 
-        public boolean start(Callback callback, boolean commit)
+        private void reset()
         {
-            State state;
+            assert Thread.holdsLock(this);
+            _state = State.IDLE;
+            _callback = null;
+            _commit = false;
+            _failure = null;
+        }
+
+        private void send(Callback callback, boolean commit, Consumer<Callback> consumer)
```

Comment:
Since all lambdas passed to this method are always capturing, I see no point in passing it the `Callback`. I think changing the signature here to accept a `Runnable` instead of a `Consumer` would be slightly more readable.

Reply:
I agree.  At the very least the param name needs to be changed to `sendFrame`.

Reply:
I like the `Consumer<Callback>` for 2 reasons:
* it allows `TransportCallback.send()` to give to the `Consumer` some other callback if needs to without having the `sendXXXFrame()` methods be aware of what callback to use
* it keeps the semantic of the `sendXXXFrame()` clear: they take a callback parameter, and therefore they are asynchronous

Reply:
@sbordet if you keep the `Consumer` it has to have a better name than `consumer`.   I think `sendFrame` is best as it indicates what the Consumer is meant to do.